backend/user_service/routes/userRoutes.js

```diff
@@ -10,8 +10,15 @@ router.post('/verify-token', authenticateToken, (req, res) => {
 });
 
 // Route to create a new user
-router.post('/user/addToUserCollection', addToUserCollection); // New route for user creation
+router.post('/user/addToUserCollection', addToUserCollection);
 
-router.get('/admin/checkAdminStatus', checkAdminStatus); // Add the route
+// Route for checking admin status
+router.get('/admin/checkAdminStatus', checkAdminStatus); 
+
+// Add the endpoint to check for username existence
+router.get('/check-username', checkUsernameExists);
+
+// Route to get username by uid
+router.get('/user/username/:uid', getUsernameByUid);
```

Comment:
Can consider adding the authenticateToken middleware for each of these routes

Reply:
Have added the middleware for all the routes except for addToUserCollection. (This one won't work as there shouldn't be any user signed in when addToUserCollection() is called. Also tried it to be sure and yea it won't work)